### PR TITLE
feat: Updte /api/content/currentuser to also return roles

### PR DESCRIPTION
**Pull request by edalex-ian**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] tests are included
- [x] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change

Quick tweak to `currentuser` endpoint to support returning `roles` so that we have it available for the `user` object in  visibility scripts.

Minor tidy-up of endpoint code too, mainly so that we can have better Swagger doco.

image

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: oeq-ts-rest-api/test/auth.test.ts [hunk 1]
```diff
@@ -26,7 +26,8 @@ test("That we're able to login", async () => {
   const userDetails = await OEQ.LegacyContent.getCurrentUserDetails(
     TC.API_PATH
   );
-  expect(userDetails).toHaveProperty('username', TC.USERNAME);
+  expect(userDetails.username).toBe(TC.USERNAME);
+  expect(userDetails.roles).toEqual(['TLE_LOGGED_IN_USER_ROLE']);
```
Comment (edalex-ian) [inline oeq-ts-rest-api/test/auth.test.ts:30]: We don't actually have a `LegacyContent.test.ts` so it made sense just to tweak this one and test the (slightly odd/unique) role is returned.

File: oeq-ts-rest-api/src/LegacyContent.ts [hunk 2]
```diff
@@ -45,6 +45,10 @@ export interface CurrentUserDetails {
   menuGroups: Array<Array<MenuItem>>;
   counts?: ItemCounts;
   canDownloadSearchResult: boolean;
+  /**
+   * UUIDs of the roles assigned to the user - as well as `TLE_LOGGED_IN_USER_ROLE` where applicable.
+   */
+  roles: string[];
```
Comment (edalex-ian) [inline oeq-ts-rest-api/src/LegacyContent.ts:51]: Note that due to `TLE_LOGGED_IN_USER_ROLE` this is a `string[]` rather than a `UuidString[]`.

**Review by PenghaiZhang — APPROVED** [review]
LGTM.

**Comment by edalex-ian** [comment]
Thanks Penghai. I have fixed a minor issue that caused the tests to fail - thankfully. And I've merged component into this so I can branch off and start on the user object support.

When this goes green, I'll merge.

**Pull request merged**